specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/ViewAttachmentFiles.tsx

```diff
@@ -0,0 +1,268 @@
+  readonly onDisambiguation:
+    | ((
+        disambiguatedId: number,
+        indexToDisambiguate: number,
+        multiple: boolean
+      ) => void)
+    | undefined;
+  readonly onFilesDropped?: (file: FileList) => void;
+}): JSX.Element {
+  const [disambiguationIndex, setDisambiguationIndex] = React.useState<
+    number | undefined
+  >(undefined);
+
+  const data = React.useMemo(
+    () =>
+      resolveAttachmentDatasetData(
+        uploadableFiles,
+        setDisambiguationIndex,
+        baseTableName
+      ),
+    [uploadableFiles, setDisambiguationIndex]
+  );
+  const headers = React.useMemo(
+    () => ({
+      selectedFileName: commonText.selectedFileName(),
+      fileSize: attachmentsText.fileSize(),
+      record: (
+        <div className="flex min-w-fit items-center gap-2">
+          {baseTableName === undefined ? (
+            formsText.record()
+          ) : (
+            <>
+              <TableIcon label name={baseTableName} />
+              {uploadSpec.staticPathKey === undefined
+                ? ''
+                : defined(
+                    strictGetModel(baseTableName).getField(
+                      staticAttachmentImportPaths[uploadSpec.staticPathKey].path
+                    )
+                  ).label}
+            </>
+          )}
+        </div>
+      ),
+      status: attachmentsText.status(),
+      attachmentId: attachmentsText.attachmentID(),
+    }),
+    [uploadSpec.staticPathKey]
+  );
+
+  const fileDropDivRef = React.useRef<HTMLDivElement>(null);
+  const { isDragging, ...restCallbacks } = useDragDropFiles(
+    handleFilesDropped,
+    fileDropDivRef
+  );
+  return (
+    <>
+      <div
+        className="shodow-md flex w-fit w-full flex-1  flex-col gap-2 overflow-auto rounded bg-[color:var(--background)] p-4"
+        {...restCallbacks}
+      >
+        <div className="font-semibold">
+          {attachmentsText.totalFiles({ fileCount: data.length })}
+        </div>
+        <div className="h-full overflow-auto" ref={fileDropDivRef}>
+          <div
+            className={`
+        grid-table
+        w-fit w-full flex-1 grid-cols-[repeat(var(--cols),auto)] rounded print:p-1
+      `}
+            role="table"
+            style={
+              { '--cols': Object.keys(headers).length } as React.CSSProperties
+            }
+          >
+            <div role="row">
+              {Object.entries(headers).map(([name, label]) => (
+                <div
+                  className={`
+              sticky top-0 border-b-2 border-gray-400 bg-[color:var(--background)]
+              p-2 font-bold dark:border-neutral-500 print:p-1 ${
+                isDragging ? 'bg-brand-100' : ''
+              }`}
+                  key={name}
+                  role="columnheader"
+                >
+                  {label}
+                </div>
+              ))}
+            </div>
+            <div role="rowgroup">
+              {data.map((row, index) => {
```

Comment:
can you format the attachment file size properly in KB/MB units? (or ideally dynamic)
use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat